src/queue.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt;
+
+// tmp allow dead_code
+#[allow(dead_code)]
+pub struct Queue<T: Clone> {
+    head: Option<Node<T>>,
+    length: i32,
+}
+
+// tmp allow dead_code
+#[allow(dead_code)]
+struct Node<T: Clone> {
+    data: T,
+    next: Option<Box<Node<T>>>,
+}
+
+// tmp allow dead_code
+#[allow(dead_code)]
+pub fn new<T: Clone>() -> Queue<T> {
+    return Queue {
+        head: None,
+        length: 0,
+    };
+}
```

Comment:
```suggestion
#[derive(Default)]
pub struct Queue<T: Clone> {
    head: Option<Node<T>>,
    length: i32,
}

// tmp allow dead_code
#[allow(dead_code)]
struct Node<T: Clone> {
    data: T,
    next: Option<Box<Node<T>>>,
}
```
The standard way to create a constructor for a type which has no args is to implement [std::default::Default](https://doc.rust-lang.org/std/default/trait.Default.html). Luckily, there's a  `derive` macro `Default that will just create the struct using the default value for each of its fields, where...
```
let q = Queue::default();
assert!(q.head.is_none());
assert_eq!(q.length, 0);
```

Reply:
Awesome I've done that